src/renderers/dom/stack/client/ReactDOMComponent.js

```diff
@@ -583,6 +584,20 @@ ReactDOMComponent.Mixin = {
         );
         didWarnShadyDOM = true;
       }
+
+      if (__DEV__) {
+        if (this._namespaceURI === DOMNamespaces.html) {
+          warning(
+            !(ownerDocument.createElement(this._tag) instanceof window.HTMLUnknownElement) ||
+              isCustomComponent(this._tag, props) || voidElementTags[this._tag],
```

Comment:
This was added to handle specific scenario of `menuitem` tag. The `<menuitem></<menuitem>' is supported only on firefox. In all other browsers the `instanceof window.HTMLUnknownElement` would throw a false positive. `menuitem` is included in `voidElementsTags` array. 

My idea was that since all items in `voidElementTags` are known elements it would be safe to check if the to be rendered element is in that array or not - because that array is guaranteed to contain only valid elements.

Maybe i can add a descriptive comment about that or do you have any other solution in mind on how to handle `menuitem` ?